compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -42,15 +43,22 @@ static bool isIdentityPermutation(ArrayRef<int64_t> perm) {
 }
 
 // Constructs a transpose of the given tensor and permutation.
-static Value createTranspose(OpBuilder &builder, Value source,
-                             ArrayRef<int64_t> perm) {
+static Value createTransposeInit(OpBuilder &builder, Value source,
+                                 ArrayRef<int64_t> perm) {
   SmallVector<OpFoldResult> mixedSizes =
       tensor::getMixedSizes(builder, source.getLoc(), source);
   applyPermutationToVector(mixedSizes, perm);
   Type elemType = cast<RankedTensorType>(source.getType()).getElementType();
   Value empty =
       builder.create<tensor::EmptyOp>(source.getLoc(), mixedSizes, elemType)
           .getResult();
+  return empty;
+}
+
+// Constructs a transpose of the given tensor and permutation.
+static Value createTranspose(OpBuilder &builder, Value source,
```

Comment:
It would be cool if you can add a builder upstream. I think `source` + `ArrayRef<int64_t> perm` provides enough information to construct the linalg transpose op.

Reply:
Sure, I can add it as a helper upstream after this sprint and at the same time as upstreaming the specialization pattern (I think this was mentioned before but I haven't gotten around to it yet and this pass is still being refined).